src/Encoder/BlockCipher.js

```diff
@@ -0,0 +1,328 @@
+
+    if (!Browser.isNode() && (modeName === 'ecb')) {
+      throw new InvalidInputError(
+        `Algorithm ${algorithmName}-${modeName} not supported by WebCrypto`)
+    }
+
+    const keyLength = BlockCipherEncoder.getAlgorithmKeySize(algorithmName) / 8
+    const blockLength = BlockCipherEncoder.getAlgorithmBlockSize(algorithmName) / 8
+
+    const key = this.getSettingValue('key')
+    if (key.length !== keyLength) {
+      throw new InvalidInputError(
+        `Algorithm ${algorithmName} requires a key of '${keyLength}' bytes`)
+    }
+
+    let iv = this.getSettingValue('iv')
+    if (BlockCipherEncoder.getModeNeedsIV(modeName)) {
+      if (iv.length !== blockLength) {
+        throw new InvalidInputError(
+          `Algorithm ${algorithmName}-${modeName} requires an IV of '${blockLength}' bytes`)
+      }
+    } else {
+      iv = undefined
+    }
+
+    const cipherText = await this.createCipher(algorithmName, modeName, key, iv, false, isEncode, content.getBytes())
+    return Chain.wrap(cipherText)
+  }
+
+  /**
+   * Creates message cipher using given algorithm.
+   * @protected
+   * @param {string} name Algorithm name
+   * @param {Uint8Array} message Message bytes
+   * @return {Promise}
+   */
+  createCipher (name, mode, key, iv, padding, isEncode, message) {
+    const algorithm = algorithms.find(algorithm => algorithm.name === name)
+
+    // console.log('ALGO: ' + name)
+    // console.log('MODE: ' + mode)
+    /* switch (name) {
+      case 'md5':
+        return new Promise(resolve => resolve(md5(message)))
+    } */
+
+    if (Browser.isNode()) {
+      const cipherName = algorithm.nodeAlgorithm + '-' + mode
+
+      // Node v8.x - convert Uint8Array to Buffer - not needed for v10
+      iv = Buffer.from(iv || [])
+
+      // create message cipher using Node Crypto async
+      return new Promise((resolve, reject) => {
+        let cipher = isEncode
+          ? nodeCrypto.createCipheriv(cipherName, key, iv)
+          : nodeCrypto.createDecipheriv(cipherName, key, iv)
+
+        cipher.setAutoPadding(padding)
+
+        const resultBuffer = Buffer.concat([cipher.update(global.Buffer.from(message)), cipher.final()])
+        resolve(new Uint8Array(resultBuffer))
+      })
+    } else {
+      // get crypto subtle instance
+      const crypto = window.crypto || window.msCrypto
+      const cryptoSubtle = crypto.subtle || crypto.webkitSubtle
+
+      const cipherName = algorithm.browserAlgorithm + '-' + mode
+      const cryptoKey = cryptoSubtle.importKey(
+        'raw',
+        key,
+        {
+          name: cipherName
+        },
+        false,
+        ['encrypt', 'decrypt']
+      )
+
+      // create message cipher using Web Crypto Api
+      let result = cryptoKey.then((keyObj) => {
+        let blockLength = algorithm.blockSize / 8
+
+        if (mode === 'ecb') {
+          // WebCrypto does not support AES-ECB
+          return Promise.resolve(new Uint8Array(blockLength))
+        }
```

Comment:
No need to add a fallback for mode `ECB` as previous checks (settings) will/should never let through this mode for the current environment, if not supported.